### ISSUE TITLE
Display potential volunteers on Allocation page

### ISSUE THREAD

**Issue by burlexpo** [bethlakshmi/GBE2#213]
Do that thing we talked about... which allows a Staff Head to make intelligent decisions when assigning volunteers.

**Comment by bethlakshmi** [comment]
Sounds about right...
- want a way to know the list of people who volunteered for something like
this
- want a way to know who is free at the time slot
- don't want to be forced into selecting only those
- reminder to everyone - volunteers are identified per profile, not per
performer

**Comment by jonkiparsky** [comment]
I've committed a partial on this - I've got dropdowns for "all volunteers" and "all volunteers who expressed an interest in this type". Styling is ugly, there's some weird div that I haven't found yet, but it's up on demo for review.

**Comment by jonkiparsky** [comment]
It's not clear to me what the time ranges for availability are. If the options are

[('SH0', 'Thursday evening'), ('SH1', 'Friday morning'),                                                                                                                                                     
                                  ('SH2', 'Friday afternoon'), ('SH3', 'Friday night'),                                                                                                                                                       
                                  ('SH4', 'Friday late night'), ('SH5', 'Saturday morning'),                                                                                                                                                  
                                  ('SH6', 'Saturday afternoon'), ('SH7', 'Saturday night'),                                                                                                                                                   
                                  ('SH8', 'Saturday late night'), ('SH9', 'Sunday morning'),                                                                                                                                                  
                                  ('SH10', 'Sunday afternoon'), ('SH11', 'Sunday night'),
                                  ('SH12', 'Strike Crew'), ('SH13', 'Monday morning')]


what do these mean in terms of concrete times of day?

**Comment by jonkiparsky** [comment]
@burlexpo - should have @mentioned you in the previous comment. Can you have a stab at answering this? It's actually not a trivial problem, but I've written a little bit of code to make it easier to solve, so I might be able to knock it out before the standup on Thursday, if I have the above information. 
(it's okay if the times overlap, by the way - in the implementation I'm thinking of, this should be fine)

**Comment by burlexpo** [comment]
@jonkiparsky -- yeah, I never saw that notification.  Here are your answers -- for clarity, when I say "8am to 1pm" I mean that the volunteer shift must finish by 1pm.  It doesn't help us to assign someone who is only available for the first hour of a shift -- does that make sense?

[('SH0', 'Thursday evening')  6pm to 11pm
('SH1', 'Friday morning'), 8am to 1pm
('SH2', 'Friday afternoon'), 1pm to 6pm
('SH3', 'Friday night'), 5pm to 10pm
('SH4', 'Friday late night'),  10pm to 1am (Saturday)
('SH5', 'Saturday morning') 8am to 1pm
('SH6', 'Saturday afternoon'),  1pm to 6pm
('SH7', 'Saturday night'),  5pm to 10pm
('SH8', 'Saturday late night'), 10pm to 1am (Sunday)
('SH9', 'Sunday morning'), 8am to 1pm
('SH10', 'Sunday afternoon'),  1pm to 6pm
('SH11', 'Sunday night'), 5pm to 10pm
('SH12', 'Strike Crew'), 10pm to 1am (Monday)
('SH13', 'Monday morning')] 8am to 12pm

**Comment by jonkiparsky** [comment]
These times are a little awkward, since any shift which spans an entire show is split across two of these. This makes it a little complicated to determine whether a volunteer is available for a show. However, I've committed some code that shows you all people who have selected a time slot which completely contains this opportunity. This might be useful.

**Comment by bethlakshmi** [comment]
Updated note:  When I did work on #463, @burlexpo and I solved the event overlaps two windows conundrum with the conclusion that it was OK to find "volunteers whose said they were available *at the start time* of this event".  We'll have to trust that the assigner and the volunteer are capable of looking at the big picture and communicating.

For reference, in volunteer assignment the logic was start time is >= volunteer window start, and < volunteer window end.
